src/App.tsx

```diff
@@ -17,18 +45,35 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                completedFilter={completedFilter}
+                setCompletedFilter={setCompletedFilter}
+                searchQuary={searchQuary}
+                setSearchQuary={setSearchQuary}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={visibleTodos}
+                  selectedTodo={selectedTodo}
+                  onSelectTodo={setSelectedTodo}
```

Comment:
The method `onSelectTodo` could be named more explicitly to reflect the action it performs. Consider renaming it to `handleTodoSelection` or similar. This would make the code easier to understand and maintain. This recommendation is based on the common naming conventions for event handler functions in React.